src/ussd_popi_faq.js

```diff
@@ -0,0 +1,329 @@
+            // initialise services
+            is = new SeedJsboxUtils.IdentityStore(
+                new JsonApi(self.im, {}),
+                self.im.config.services.identity_store.token,
+                self.im.config.services.identity_store.url
+            );
+            sbm = new SeedJsboxUtils.StageBasedMessaging(
+                new JsonApi(self.im, {}),
+                self.im.config.services.stage_based_messaging.token,
+                self.im.config.services.stage_based_messaging.url
+            );
+            hub = new SeedJsboxUtils.Hub(
+                new JsonApi(self.im, {}),
+                self.im.config.services.hub.token,
+                self.im.config.services.hub.url
+            );
+            ms = new SeedJsboxUtils.MessageSender(
+                new JsonApi(self.im, {}),
+                self.im.config.services.message_sender.token,
+                self.im.config.services.message_sender.url,
+                self.im.config.services.message_sender.channel
+            );
+
+            self.env = self.im.config.env;
+            self.metric_prefix = [self.env, self.im.config.name].join('.');
+            self.store_name = [self.env, self.im.config.name].join('.');
+
+            self.attach_session_length_helper(self.im);
+
+            mh = new MetricsHelper(self.im);
+            mh
+                // Total unique users
+                // This adds <env>.ussd_public.sum.unique_users 'last' metric
+                // as well as <env>.ussd_public.sum.unique_users.transient 'sum' metric
+                .add.total_unique_users([self.metric_prefix, 'sum', 'unique_users'].join('.'))
+
+                // Total sessions
+                // This adds <env>.ussd_public.sum.sessions 'last' metric
+                // as well as <env>.ussd_public.sum.sessions.transient 'sum' metric
+                .add.total_sessions([self.metric_prefix, 'sum', 'sessions'].join('.'))
+
+                // Total unique users for environment, across apps
+                // This adds <env>.sum.unique_users 'last' metric
+                // as well as <env>.sum.unique_users.transient 'sum' metric
+                .add.total_unique_users([self.env, 'sum', 'unique_users'].join('.'))
+
+                // Total sessions for environment, across apps
+                // This adds <env>.sum.sessions 'last' metric
+                // as well as <env>.sum.sessions.transient 'sum' metric
+                .add.total_sessions([self.env, 'sum', 'sessions'].join('.'))
+
+                // Average sessions to register
+                .add.tracker({
+                    action: 'exit',
+                    state: 'state_start'
+                }, {
+                    action: 'enter',
+                    state: 'state_end_success'
+                }, {
+                    sessions_between_states: [self.metric_prefix, 'avg.sessions_to_register'].join('.')
+                })
+            ;
+
+            self.im.on('state:exit', function(e) {
```

Comment:
you probably don't need this method or `fire_complete`. It just fires a metric to record how many people completed the registration. Since we never exit and it's up to the user to close the session, I don't think this should ever get called